### PR TITLE
feat(instrumentation-dbapi): add experimental sql commenter capability

### PR DESCRIPTION
**Pull request by lonewolf3739**
# Description

This PR adds support for an experimental and optional feature for enabling sql commenter capability to dbapi instrumentation. 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] `tox -e test-instrumentation-dbapi`

# Does This PR Require a Core Repo Change?

- [x] No.

### CODE REVIEW

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py [hunk 1]
```diff
@@ -115,3 +116,38 @@ def _start_internal_or_server_span(
         attributes=attributes,
     )
     return span, token
+
+
+_KEY_VALUE_DELIMITER = ","
+
+
+def _generate_sql_comment(**meta):
```
Comment (ocelotl) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:124]: Is it really necessary to add this to `utils`? This is a single-use function, its contents can just be used directly where the function is called.

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py [hunk 2]
```diff
@@ -115,3 +116,38 @@ def _start_internal_or_server_span(
         attributes=attributes,
     )
     return span, token
+
+
+_KEY_VALUE_DELIMITER = ","
```
Comment (ocelotl) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:121]: Single-use constant

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 3]
```diff
@@ -374,6 +397,16 @@ def traced_execution(
...
+            if args and self._commenter_enabled:
+                try:
+                    comment = self._generate_comment(span)
+                    if isinstance(args[0], bytes):
+                        comment = comment.encode("utf8")
+                    args_list = list(args)
+                    args_list[0] += comment
+                    args = tuple(args_list)
+                except Exception as exc:
+                    _logger.warning("Exception while generating sql comment: %s", exc)
```
Comment (ocelotl) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:409]:
```suggestion
                    _logger.exception("Exception while generating sql comment: %s", exc)
```
Is [this](https://docs.python.org/3/library/logging.html#logging.Logger.exception) more appropriate?

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py (hunk 1 above)
Comment (lonewolf3739) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py:124]: I think so, there will be many sql db instrumentations that will rely on this.

**Review by ocelotl — APPROVED** [review]

**Pull request merged**